### PR TITLE
[IOTDB-3718] Unify retry logic of SyncClientPool in ConfigNode

### PR DESCRIPTION
**Pull request by RYH61**
Unify retry logic of SyncClientPool in ConfigNode
see details : https://issues.apache.org/jira/browse/IOTDB-3718

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/client/SyncConfigNodeClientPool.java [hunk 1]
```diff
@@ -62,118 +65,74 @@ private void updateConfigNodeLeader(TSStatus status) {
...
+            requestType + " failed on ConfigNode {}, because {}, retrying {}...",
+            endPoint,
+            e.getMessage(),
+            retry);
         doRetryWait(retry);
       }
     }
-
-    throw lastException;
+    LOGGER.error(requestType + " failed on ConfigNode {}", endPoint, lastException);
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/client/SyncConfigNodeClientPool.java:102]: requestType  user {}

File: confignode/src/main/java/org/apache/iotdb/confignode/client/SyncConfigNodeClientPool.java [hunk 2]
```diff
@@ -62,118 +65,74 @@ private void updateConfigNodeLeader(TSStatus status) {
...
+            return client.stopConfigNode((TConfigNodeLocation) req);
+          default:
+            return RpcUtils.getStatus(
+                TSStatusCode.EXECUTE_STATEMENT_ERROR, "Unknown request type: " + requestType);
+        }
+      } catch (Throwable e) {
         lastException = e;
         LOGGER.warn(
-            "Add Consensus Group failed because {}, retrying {} ...", e.getMessage(), retry);
+            requestType + " failed on ConfigNode {}, because {}, retrying {}...",
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/client/SyncConfigNodeClientPool.java:95]: requestType  user {}

File: confignode/src/main/java/org/apache/iotdb/confignode/client/SyncDataNodeClientPool.java [hunk 3]
```diff
@@ -54,35 +58,37 @@ private SyncDataNodeClientPool() {
...
+          case invalidatePermissionCache:
+            return client.invalidatePermissionCache((TInvalidatePermissionCacheReq) req);
+          default:
+            return RpcUtils.getStatus(
+                TSStatusCode.EXECUTE_STATEMENT_ERROR, "Unknown request type: " + requestType);
+        }
+      } catch (TException | IOException e) {
+        lastException = e;
+        LOGGER.warn(
+            requestType + " failed on DataNode {}, because {}, retrying {}...",
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/client/SyncDataNodeClientPool.java:82]: the same as above

File: confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeRemoveCheck.java [hunk 4]
```diff
@@ -72,14 +73,24 @@ public TConfigNodeLocation removeCheck(TEndPoint endPoint) {
...
+    TSStatus status = new TSStatus();
+    for (TConfigNodeLocation configNodeLocation : getConfigNodeList()) {
+      status =
+          (TSStatus)
+              SyncConfigNodeClientPool.getInstance()
+                  .sendSyncRequestToConfigNode(
+                      configNodeLocation.getInternalEndPoint(),
+                      nodeLocation,
+                      ConfigNodeRequestType.removeConfigNode);
+      break;
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/conf/ConfigNodeRemoveCheck.java:85]: if （status  == success ） {break}.       

when failed, will use next  configNodeLocation .

File: confignode/src/main/java/org/apache/iotdb/confignode/client/SyncDataNodeClientPool.java [hunk 5]
```diff
@@ -54,35 +58,37 @@ private SyncDataNodeClientPool() {
...
-    TSStatus status;
-    try (SyncDataNodeInternalServiceClient client = clientManager.borrowClient(endPoint)) {
-      status = client.invalidateSchemaCache(invalidateCacheReq);
-    } catch (IOException e) {
-      LOGGER.error("Can't connect to DataNode {}", endPoint, e);
-      status = new TSStatus(TSStatusCode.TIME_OUT.getStatusCode());
-    } catch (TException e) {
-      LOGGER.error("Invalid Schema Cache on DataNode {} failed", endPoint, e);
-      status = new TSStatus(TSStatusCode.EXECUTE_STATEMENT_ERROR.getStatusCode());
+  public Object sendSyncRequestToDataNode(
```
Comment (wangchao316) [inline confignode/src/main/java/org/apache/iotdb/confignode/client/SyncDataNodeClientPool.java:61]: Object alter to TSStatus

**Review by wangchao316 — CHANGES_REQUESTED** [review]

**Review by wangchao316 — APPROVED** [review]
LGTM

**Review by CRZbulabula — APPROVED** [review]
LGTM

**Pull request merged**